### ISSUE TITLE
Add a subcommands to check for the .swiftlint.yml syntax and path

### ISSUE THREAD

**Issue by AliSoftware** [realm/SwiftLint#533]
Once #532 is accepted and merged, another possible action for the `swiftlint config` subcommand might be `swiftlint config --check` to check the syntax of the `.swiftlint.yml`

This action would ensure that there is no unknown key being used (like a typo in a key or a key which would have been removed or renamed between SwiftLint versions, etc)
It would also ensure that the keys have the expected type. e.g. I've been bitten multiple times by using the wrong yaml config file like this:

``` yaml
line_length:
 - warning: 120
 - error: 150
```

Note that the extra dashes make `line_length` and array containing two hashes, each with only one key, instead of making `line_length` a single hash with two keys. `swiftlint config --check` might help spotting the typo.

---

Other subcommands worth adding could include:
- `swiftlint config --path` to identify all the `.swiftlint.yml` files which would be considered if we ran `swiftlint lint` in the current directory
- `swiftlint config --edit` to launch the current `$EDITOR` with the first `.swiftlint.yml` file found
